geode-core/src/main/java/org/apache/geode/cache/HttpService.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.internal.cache.InternalCache;
+
+/**
+ * This interface provides access to the simple http service exposed on the Geode
+ * {@link InternalCache}. An instance can be retrieved by calling
+ * {@code InternalCache.getHttpService()}.
+ */
+@Experimental
+public interface HttpService {
+
+  /**
+   * Stop the service. This should also stop any transitively managed components. Starting the
+   * service is implicitly handled by {@link #addWebApplication(String, String, Pair[])}
+   */
+  void stop();
```

Comment:
Seems dangerous to have a global stop method on this API. This should only be executed by the internal lifecycle when shutting down. See below for comment on per-context lifecycle.